File: todo_frontend/lib/todo/models/todo.dart
class Todo {
  final String id;
  final String title;
  final bool completed;

  Todo({required this.id, required this.title, required this.completed});

  factory Todo.fromJson(Map<String, dynamic> json) {
    return Todo(
      id: json['ID'].toString(),
      title: json['title'],
      completed: json['completed'],
    );
  }

  Map<String, dynamic> toJson() => {
    'id': id,
    'title': title,
    'completed': completed,
  };
}

File: todo-backend/models/Todo.go
package models

import (
	"gorm.io/gorm"
)

// we create a type called Todos which is a slice of Todo
type Todos []Todo

type Todo struct {
	gorm.Model
	Title     string `json:"title"`
	Completed bool   `json:"completed"`
}

func NewTodo(title string) *Todo {
	return &Todo{Title: title, Completed: false}
}
